core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -509,11 +474,14 @@ protected ReadOnlyKeyCommand remoteReadOnlyCommand(InvocationContext ctx, ReadOn
       return new TxReadOnlyKeyCommand(command, getMutationsOnKey((TxInvocationContext) ctx, command.getKey()));
    }
 
+   protected RpcOptions createRpcOptions() {
+      return rpcManager.getRpcOptionsBuilder(ResponseMode.SYNCHRONOUS_IGNORE_LEAVERS, DeliverOrder.NONE).build();
```

Comment:
Eh, {{RpcOptions}} are cached on a handful of places - OK, I am the villain who does that, but noone complained during review so far. Does it make sense to change that timeout? The builder + the RpcOptions is created thousand times per second, there should be at least an event to refresh such cached objects.